addons/im_livechat/static/src/models/public_livechat_global_notification_handler.js

```diff
@@ -0,0 +1,102 @@
+                        page_history: history,
+                    });
+                    return;
+                }
+                case 'mail.channel.member/typing_status': {
+                    if (payload.channel_id !== this.messaging.livechatButtonView.publicLivechat.legacyPublicLivechat._id) {
+                        return;
+                    }
+                    const partnerID = payload.partner_id;
+                    if (partnerID === this.messaging.livechatButtonView.currentPartnerId) {
+                        // ignore typing display of current partner.
+                        return;
+                    }
+                    if (payload.is_typing) {
+                        this.messaging.livechatButtonView.publicLivechat.legacyPublicLivechat.registerTyping({ partnerID });
+                    } else {
+                        this.messaging.livechatButtonView.publicLivechat.legacyPublicLivechat.unregisterTyping({ partnerID });
+                    }
+                    return;
+                }
+                case 'mail.channel/new_message': {
+                    if (payload.id !== this.messaging.livechatButtonView.publicLivechat.legacyPublicLivechat._id) {
+                        return;
+                    }
+                    const notificationData = payload.message;
+                    // If message from notif is already in chatter messages, stop handling
+                    if (this.messaging.livechatButtonView.messages.some(message => message.getID() === notificationData.id)) {
+                        return;
+                    }
+                    notificationData.body = utils.Markup(notificationData.body);
+                    this.messaging.livechatButtonView.widget._addMessage(notificationData);
+                    if (this.messaging.livechatButtonView.chatWindow.legacyChatWindow._thread._folded || !this.messaging.livechatButtonView.chatWindow.legacyChatWindow._publicLivechatView.isAtBottom()) {
+                        this.messaging.livechatButtonView.publicLivechat.legacyPublicLivechat._unreadCounter++;
+                    }
+                    this.messaging.livechatButtonView.widget._renderMessages();
+                    return;
+                }
+                case 'mail.message/insert': {
+                    const message = this.messaging.livechatButtonView.messages.find(message => message._id === payload.id);
+                    if (!message) {
+                        return;
+                    }
+                    message._body = utils.Markup(payload.body);
+                    this.messaging.livechatButtonView.widget._renderMessages();
+                    return;
+                }
+            }
+        },
+        /**
+         * @private
+         * @param {Array[]} notifications
+         */
+        _onNotification(notifications) {
+            for (let notification of notifications) {
```

Comment:
`let` -> `const`? You probably don't want to make any changes other than the move, but on the other hand you probably won't have many opportunities to make diff on these lines later. It could be now or never.